openassessment/xblock/staff_assessment_mixin.py

```diff
@@ -0,0 +1,119 @@
+    """
+    This mixin is for all staff-assessment related endpoints.
+    """
+
+    def staff_assessment_exists(self, submission_uuid):
+        """
+        Returns True if there exists a staff assessment for the given uuid. False otherwise.
+        """
+
+        return staff_api.get_latest_staff_assessment(submission_uuid) is not None
+
+    @XBlock.json_handler
+    @require_course_staff("STUDENT_INFO")
+    def staff_assess(self, data, suffix=''):
+        """
+        Create a staff assessment from a staff submission.
+        """
+        if 'options_selected' not in data:
+            return {'success': False, 'msg': self._(u"Missing options_selected key in request")}
+
+        if 'overall_feedback' not in data:
+            return {'success': False, 'msg': self._('Must provide overall feedback in the assessment')}
+
+        if 'criterion_feedback' not in data:
+            return {'success': False, 'msg': self._('Must provide feedback for criteria in the assessment')}
+
+        if 'submission_uuid' not in data:
+            return {'success': False, 'msg': self._(u"Missing the submission id of the submission being assessed.")}
+
+        try:
+            assessment = staff_api.create_assessment(
+                data['submission_uuid'],
+                self.get_student_item_dict()["student_id"],
+                data['options_selected'],
+                clean_criterion_feedback(self.rubric_criteria, data['criterion_feedback']),
+                data['overall_feedback'],
+                create_rubric_dict(self.prompts, self.rubric_criteria_with_labels)
+            )
+            self.publish_assessment_event("openassessmentblock.staff_assessment", assessment)
+            workflow_api.update_from_assessments(assessment["submission_uuid"], {}, force_update_score=True)
+
+        except StaffAssessmentRequestError:
+            logger.warning(
+                u"An error occurred while submitting a staff assessment "
+                u"for the submission {}".format(self.submission_uuid),
+                exc_info=True
+            )
+            msg = self._(u"Your staff assessment could not be submitted.")
+            return {'success': False, 'msg': msg}
+        except StaffAssessmentInternalError:
+            logger.exception(
+                u"An error occurred while submitting a staff assessment "
+                u"for the submission {}".format(self.submission_uuid),
+            )
+            msg = self._(u"Your staff assessment could not be submitted.")
+            return {'success': False, 'msg': msg}
+        else:
+            return {'success': True, 'msg': u""}
+
+    @XBlock.handler
+    @require_course_staff("STUDENT_INFO")
+    def render_staff_assessment(self, data, suffix=''):
+        """
+        Render the staff assessment for the given student.
+        """
+        try:
+            submission_uuid = data.get("submission_uuid")
+            path, context = self.self_path_and_context(submission_uuid)
```

Comment:
The best way to understand `self.submission_uuid` is to understand that it's the `submission_uuid` of the current xblock, which is matched up with the submission of the current user who's viewing the xblock. The reason why we want to fetch a separate `submission_uuid` in the staff mixin methods is because the staff member is acting on any student's submission, not just their own.